server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -387,4 +388,33 @@ private void reject(ThreadPoolExecutor executor, Runnable task) {
         }
     }
 
+    public static class TaskTrackingConfig {
+        private final boolean trackExecutionTime;
+        private final double ewmaAlpha;
+
+        public static TaskTrackingConfig DO_NOT_TRACK = new TaskTrackingConfig(false);
+        public static TaskTrackingConfig DEFAULT = new TaskTrackingConfig(true);
+
+        public TaskTrackingConfig(double EWMAAlpha) {
+            this(true, EWMAAlpha);
+        }
+
+        private TaskTrackingConfig(boolean trackExecutionTime) {
```

Comment:
nit: could the constant definitions above use the ctor taking both the alpha and boolean? Having two single arg ctors, one public one private, is a little confusing.